src/pseudopeople/interface.py

```diff
@@ -182,7 +182,7 @@ def _generate_dataset(
             noised_dataset = dask_data.map_partitions(
                 lambda data, partition_info=None: noise_data(
                     dataset_schema,
-                    data,
+                    data.copy(),
```

Comment:
does this mean we just increased the memory requirements?

Reply:
I'm not sure. If we pass in data.copy() but it gets modified inplace and so data.copy() turns into noised_dataset then no. At least thats how Im thinking about it